supervisor/resolution/checks/detached_addon.py

```diff
@@ -0,0 +1,46 @@
+"""Helpers to check for detached addons."""
+
+from ...const import CoreState
+from ...coresys import CoreSys
+from ..const import ContextType, IssueType
+from .base import CheckBase
+
+
+def setup(coresys: CoreSys) -> CheckBase:
+    """Check setup function."""
+    return CheckDetachedAddon(coresys)
+
+
+class CheckDetachedAddon(CheckBase):
+    """CheckDetachedAddon class for check."""
+
+    async def run_check(self) -> None:
+        """Run check if not affected by issue."""
+        for addon in self.sys_addons.installed:
+            if addon.is_detached:
+                self.sys_resolution.create_issue(
+                    IssueType.DETACHED_ADDON,
+                    ContextType.ADDON,
+                    reference=addon.slug,
+                )
+
+    async def approve_check(self, reference: str | None = None) -> bool:
+        """Approve check if it is affected by issue."""
+        return (
+            addon := self.sys_addons.get(reference, local_only=True)
+        ) and addon.is_detached
+
+    @property
+    def issue(self) -> IssueType:
+        """Return a IssueType enum."""
+        return IssueType.DETACHED_ADDON
+
+    @property
+    def context(self) -> ContextType:
+        """Return a ContextType enum."""
+        return ContextType.ADDON
+
+    @property
+    def states(self) -> list[CoreState]:
+        """Return a list of valid states when this check can run."""
+        return [CoreState.RUNNING, CoreState.SETUP]
```

Comment:
```suggestion
        return [CoreState.STARTUP, CoreState.RUNNING]
```
At this point, addons are loaded, SETUP can be to early I feel. I guess we need RUNNING as well detect detach add-ons after store updates

Reply:
@pvizeli we can do either. Currently the healthcheck during `SETUP` is at the very end after everything else in that step so all addon information is loaded:
https://github.com/home-assistant/supervisor/blob/67ec89c9567016182d77d22968cb9632820f091f/supervisor/core.py#L191

I just removed `RUNNING` from there as per @agners request although I can add it back if we want. Once at startup of supervisor seemed like enough, this should be uncommon other then possibly the source code of a local add-on being removed so I don't think we need to catch this immediately.